File: frontend/src/types/equipment.ts
import type { Category } from "./category";

export interface Equipment {
  idEquipment: number;
  category: Category;
  brand: string;
  model: string;
  year?: number;
}

// Si besoin plus tard :
export interface EquipmentCreate {
  category: {
    idCategory: number;
  };
  brand: string;
  model: string;
  year?: number;
}

File: backend/src/main/java/com/solvhub/model/Equipment.java
package com.solvhub.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "equipment")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Equipment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idEquipment;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "idCategory", nullable = false)
    private Category category;

    @Column(nullable = false)
    private String brand;

    @Column(nullable = false)
    private String model;

    @Column(name = "year")
    private Integer year;
}
